internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -2544,3 +2559,55 @@ resource "azurerm_kubernetes_cluster_node_pool" "test" {
 }
  `, data.Locations.Primary, data.RandomInteger)
 }
+
+func (KubernetesClusterNodePoolResource) sourceSnapshotId(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-aks-%[2]d"
+  location = "%[1]s"
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%[2]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%[2]d"
+  default_node_pool {
+    name       = "default"
+    node_count = 1
+    vm_size    = "Standard_D2s_v3"
+  }
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_kubernetes_cluster_node_pool" "source" {
+  name                  = "source"
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.test.id
+  vm_size               = "Standard_D2s_v3"
+  provisioner "local-exec" {
+    command = "az aks nodepool snapshot create --name %[3]s --resource-group ${azurerm_resource_group.test.name} --nodepool-id ${self.id} --location ${azurerm_resource_group.test.location}"
+  }
+}
+
+resource "azurerm_kubernetes_cluster_node_pool" "test" {
+  name                  = "new"
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.test.id
+  vm_size               = "Standard_DS2_v2"
+  node_count            = 1
+  source_snapshot_id    = "${azurerm_resource_group.test.id}/providers/Microsoft.ContainerService/snapshots/%[3]s"
```

Comment:
Whilst I don't think there's much value in having a resource for node pool snapshots, I think a data source would make sense so that the ID can be retrieved instead of needing to be constructed manually by the user.